source/robot_model/include/robot_model/Model.hpp

```diff
@@ -210,21 +233,58 @@ class Model {
                                                                     const std::vector<std::string>& frame_names);
 
   /**
- * @brief Compute the forward geometry, i.e. the pose of the frame from the joint values
- * @param joint_state the joint state of the robot
- * @param frame_name name of the frame at which we want to extract the pose
- * @return the pose of the desired frame
- */
+   * @brief Compute the forward geometry, i.e. the pose of the frame from the joint values
+   * @param joint_state the joint state of the robot
+   * @param frame_name name of the frame at which we want to extract the pose
+   * @return the pose of the desired frame
+   */
   state_representation::CartesianPose forward_geometry(const state_representation::JointState& joint_state,
                                                        std::string frame_name = "");
 
+
+  /**
+   * @brief parameters for the inverse geometry
+   * @param damp damping added to the diagonal of J*Jt in order to avoid the singularity
+   * @param alpha alpha €]0,1], it is used to make Newthon-Raphson method less aggressive
+   * @param gamma gamma €]0,1], represents the strength of the repulsive potential field in the Clamping Weighted Least-Norm method
+   * @param margin the distance from the joint limit on which we want to penalize the joint position
+   * @param tolerance the maximum error tolerated between the desired cartesian state and the one obtained by the returned joint positions
+   * @param max_number_of_iterations the maximum number of iterations that the algorithm do for solving the inverse geometry
+   */
+  struct inverse_geometry_parameters
+  {
+      double damp;
+      double alpha;
+      double gamma;
+      double margin;
+      double tolerance;
+      int max_number_of_iterations;
+  };
+ 
+  /**
+   * @brief Compute the inverse geometry, i.e. joint values from the pose of the end-effector in a iteratively manner
+   * @param desired_cartesian_state containing the desired pose of the end-effector
+   * @param frame_name name of the frame at which we want to extract the pose
+   * @param inverse_geometry_parameters set of parameters for tunning the algorithm
+   * @return the joint positions of the robot
+   */
+  state_representation::JointPositions inverse_geometry(const state_representation::CartesianState& desired_cartesian_state,
+                                                            std::string frame_name="",
+                                                            inverse_geometry_parameters params = {1e-6, 0.8, 0.75, 0.07, 1e-3, 1000});
```

Comment:
Thank you for the review.

It seams not possible to use the aggregate initialization `params ={}`if there are default parameters defined in the structure.
https://stackoverflow.com/questions/37776823/could-not-convert-from-brace-enclosed-initializer-list-to-struct

and about the frame name it can not be constant because we modify its value, like in the forward_geometry:
```forward_geometry(const state_representation::JointState& joint_state,
                                                       std::string frame_name = "");```

This I can do `const InverseGeometryParameters&`

Reply:
I disagree, this should work in all cases as it is defined behaviour. The empty bracket `{}` prompts the empty value initialisation, which calls a default constructor as long as it is trivial. In this case, the default constructor is trivial and the values would be assigned normally.

The link you shared specifically referenced the non-default case, where the bracket initialisation contains real values (e.g. aggregate initialisation for member variables), which is different from the empty case.

But, if you want to be more clear about it, you can always explicitly call the default constructor as
```c++
const InverseGeometryParameters& params = InverseGeometryParameters()
```

I stand by the fact that the default values should _not_ be literal the function signature.